### PR TITLE
[ROCm] Enable RCCL in XLA pjrt

### PR DESCRIPTION
**Pull request by rahulbatra85**
Enables RCCL support in XLA pjrt for ROCm

### CODE REVIEW

File: tensorflow/compiler/xla/pjrt/gpu/BUILD [hunk 1]
```diff
@@ -51,8 +51,8 @@ cc_library(
         "//tensorflow/compiler/xla/stream_executor:device_mem_allocator",
         "//tensorflow/compiler/xla/stream_executor:device_memory",
         "//tensorflow/compiler/xla/stream_executor:tf_allocator_adapter",
-    ] + if_cuda([
         ":nccl_id_store",
```
Comment (jayfurmanek) [inline tensorflow/compiler/xla/pjrt/gpu/BUILD]: nit: can you move this to the top (under `:gpu_helpers`)

**Review by jayfurmanek — APPROVED** [review]
Looks good!

File: tensorflow/compiler/xla/pjrt/gpu/BUILD (hunk 1 above)
Comment (rahulbatra85) [reply tensorflow/compiler/xla/pjrt/gpu/BUILD]: done

**Comment by rahulbatra85** [comment]
retest Ubuntu-GPU-multi please!

**Pull request merged**